Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,46 @@
-function setAlarm() {}
+let timeRemaining;
+let timer;
+
+document.getElementById("set").addEventListener("click", () => {
+  setAlarm();
+});
+
+document.getElementById("stop").addEventListener("click", () => {
+  clearInterval(timer);
+  document.getElementById("set").innerText = "Set Alarm";
+  document.getElementById("timeRemaining").innerText = "Time Remaining: 00:00";
+  document.getElementById("alarmSet").value = "";
+  audio.pause();
+  audio.currentTime = 0;
+});
+
+function setAlarm() {
+  timeRemaining = parseInt(document.getElementById("alarmSet").value, 10);
+  if (isNaN(timeRemaining) || timeRemaining <= 0) {
+    alert("Please enter a valid positive number");
+    return;
+  }
+  clearInterval(timer); 
+  startTimer(); 
+}
+
+function startTimer() {
+  timer = setInterval(() => {
```

Comment:
If I put 1 in the box, and press "Set Alarm", the sound doesn't actually start for 2 seconds, rather than 1. Can you work out why this is, and fix it?